src/error.rs

```diff
@@ -26,6 +26,14 @@ pub enum Error {
     #[error("illegal argument: {0}")]
     IllegalArgument(String),
 
+    /// Generic error for HTTP connections.
+    #[error("http: {0}")]
+    Http(String),
```

Comment:
Why not propagate the error from the `http` crate?

Reply:
http::Error and hyper::Error do not implement PartialEq or Eq

Reply:
Do we still need `PartialEq` and `Eq` if we're using `assert_matches`? Are there other places we need equality?